tests/integration/targets/helm_registry_auth/tasks/tests.yml

```diff
@@ -0,0 +1,114 @@
+---
+# testing login
+- name: login to remote registry (check_mode)
+  kubernetes.core.helm_registry_auth:
+    state: present
+    host: "localhost:{{ registry_port }}"
+    validate_certs: false
+    username: testuser
+    password: pass123!
+    binary_path: "{{ helm_binary_path }}"
+  register: _login_check_mode
+  check_mode: true
+
+- assert:
+    that:
+      - _login_check_mode is changed
+
+- name: trying to install chart with login in check_mode
+  kubernetes.core.helm:
+    binary_path: "{{ helm_binary_path }}"
+    name: "test-chart-registry"
+    chart_ref: "oci://localhost:{{ registry_port }}/helm-charts/python-app"
+    chart_version: 0.1.0
+    namespace: "{{ test_namespace[0] }}"
+    create_namespace: true
+  ignore_errors: true
+  register: install_fail
+
+- name: Validate module failure
+  assert:
+    that:
+      - install_fail is failed
+
+- name: login to remote registry
+  kubernetes.core.helm_registry_auth:
+    state: present
+    host: "localhost:{{ registry_port }}"
+    validate_certs: false
+    username: testuser
+    password: pass123!
+    binary_path: "{{ helm_binary_path }}"
+  register: _login
+
+- assert:
+    that:
+      - _login is changed
+
+- name: Install chart should succeed now since login was performed
+  kubernetes.core.helm:
+    binary_path: "{{ helm_binary_path }}"
+    name: "test-chart-registry"
+    chart_ref: "oci://localhost:{{ registry_port }}/helm-charts/python-app"
+    chart_version: 0.1.0
+    namespace: "{{ test_namespace[0] }}"
+    create_namespace: true
```

Comment:
Shouldn't we verify the success of this task to ensure that the login is, indeed, successful?